### PR TITLE
Sol Gov Representative, Sol Federal Agent & APA removal

### PR DESCRIPTION
**Pull request by Mannybrado**
Reimplements Sol Gov Representative.
It has two alt-titles, Diplomat and Negotiator.
They are basically the supreme law without the ability to enforce it. When the Captain does something bad, this guy is the guy you go to. Otherwise, you now have a job to play when you want to slap people for acting like teenagers aboard the ship or to be a sleazy shithead and go "no!" with your suit jacket.

A brand new role, Sol Federal Agent. Basically the SGR's bodyguard, they give the rep a little more power to do things. 
It has two alt-titles, Assistant and Chaperone.
Now YOU can be CIA Agent from Crashpoint. Well, you could before, but...

image
oh yeah baby. (i've since added a solgov table flag on the table and a big one behind them)
Disables the Corporate Representative's bodyguard. The CR's room is now the SGR's room, changed up a small bit. The Corp Rep now resides in the "VIP" room, the one to the left of the psi advisor's office.
Finally, disables symbiote from the job list, finally.
TM first please

### CODE REVIEW

**Comment by Mannybrado** [comment]
corp rep re-enabled. did you know they never spawned in their office? i never thought about that when i mained APA. oh well!

File: code/game/objects/items/weapons/cards_ids.dm [hunk 1]
```diff
@@ -592,7 +592,7 @@ var/const/NO_EMAG_ACT = -50
 /obj/item/weapon/card/id/civilian/chaplain
 	job_access_type = /datum/job/chaplain
 
-/obj/item/weapon/card/id/civilian/head //This is not the HoP. There's no position that uses this right now.
```
Comment (Kaksisilma) [inline code/game/objects/items/weapons/cards_ids.dm]: This is clearer. I suggest keeping it if this goes through.

File: maps/torch/job/command_jobs.dm [hunk 2]
```diff
@@ -369,3 +369,51 @@
 /datum/job/bridgeofficer/get_description_blurb()
 	return "You are a Bridge Officer. You are a very junior officer. You do not give orders of your own. You are subordinate to all of command. You handle matters on the bridge and report directly to the CO and XO. You take the Dagon's helm and pilot the Byakhee if needed. You monitor bridge computer programs and communications and report relevant information to command."
 
+/datum/job/solrep
+	title = "Sol Gov Representative"
+	department = "Command"
+	department_flag = COM
```
Comment (Kaksisilma) [inline maps/torch/job/command_jobs.dm]:
```suggestion
	department_flag = SPT
```

File: maps/torch/job/command_jobs.dm [hunk 3]
```diff
@@ -369,3 +369,51 @@
 /datum/job/bridgeofficer/get_description_blurb()
 	return "You are a Bridge Officer. You are a very junior officer. You do not give orders of your own. You are subordinate to all of command. You handle matters on the bridge and report directly to the CO and XO. You take the Dagon's helm and pilot the Byakhee if needed. You monitor bridge computer programs and communications and report relevant information to command."
 
+/datum/job/solrep
+	title = "Sol Gov Representative"
+	department = "Command"
```
Comment (Kaksisilma) [inline maps/torch/job/command_jobs.dm]:
```suggestion
	department = "Support"
```

File: maps/torch/job/command_jobs.dm [hunk 4]
```diff
@@ -369,3 +369,51 @@
...
+	department_flag = COM
+	total_positions = 1
+	spawn_positions = 1
+	supervisors = "Central Command and the Sol Code of Military Justice"
+	selection_color = "#2f2f7f"
+	minimal_player_age = 14
+	economic_power = 20
```
Comment (Kaksisilma) [inline maps/torch/job/command_jobs.dm]:
```suggestion
	economic_power = 16 // same as commanding officer's
```

File: maps/torch/job/command_jobs.dm [hunk 5]
```diff
@@ -369,3 +369,51 @@
...
+	max_skill = list(   SKILL_BUREAUCRACY       = SKILL_MAX,
+	                    SKILL_FINANCE     = SKILL_MAX)
+	skill_points = 20
+
+	software_on_spawn = list(/datum/computer_file/program/comm,
+							 /datum/computer_file/program/card_mod,
+							 /datum/computer_file/program/camera_monitor,
+							 /datum/computer_file/program/reports)
+
+	access = list(access_adjudicator, access_security, access_brig, access_forensics_lockers, access_heads, access_medical, access_morgue,
```
Comment (Kaksisilma) [inline maps/torch/job/command_jobs.dm]: Hard no on this absurd amount of access. Here's a suggestion:
access = list(
		access_representative, access_security, access_medical,
		access_bridge, access_cargo, access_solgov_crew,
		access_hangar, access_torch_fax
	)

File: maps/torch/job/command_jobs.dm [hunk 6]
```diff
@@ -369,3 +369,51 @@
...
+					access_emergency_armory, access_sec_guard, access_gun, access_expedition_shuttle, access_guppy, access_seneng, access_senmed, access_senadv,
+					access_explorer, access_pathfinder, access_pilot, access_commissary, access_petrov, access_petrov_helm, access_petrov_analysis, access_petrov_phoron,
+					access_petrov_toxins, access_petrov_chemistry, access_petrov_security, access_petrov_maint, access_rd, access_petrov_rd)
+
+/datum/job/solrep/get_description_blurb()
+	return "You are the Sol Gov Representative. You are subordinate only to Central Command. Your job is represent the interests of the Solarian Central Government, to maintain said interests, and to check and recheck the ship's conduct. You use your immense legal and diplomatic power to keep any unlawful individual from escaping justice; You are able to approve, oversee, overrule any legal action aboard the ship. Remember: you hold incredible responsibility as the direct representative of the Solarian Central Govermnent and that no one on the ship holds the same kind of power you do... yet you are only one person."
```
Comment (Kaksisilma) [inline maps/torch/job/command_jobs.dm]: I feel like this is a less aggressive blurb. Keep in mind that they are a representative and a bureaucrat, not a Chief of Security. This is supposed to be a RP role more than anything.
```suggestion
	return "You are the Sol Gov Representative. You are a civilian assigned as both a diplomatic liaison for first contact and foreign affair situations on board. You are also responsible for monitoring for any serious missteps of justice, sol law or other ethical or legal issues aboard and informing and advising the Commanding Officer of them. You are a mid-level bureaucrat. You liaise between the crew and governmental interests on board. Send faxes back to Sol on mission progress and important events."
```

File: maps/torch/job/corporate_jobs.dm [hunk 7]
```diff
@@ -42,8 +42,8 @@
 	title = "Loss Prevention Associate"
 	department = "Support"
 	department_flag = SPT
-	total_positions = 1
-	spawn_positions = 1
+	total_positions = 0
```
Comment (Kaksisilma) [inline maps/torch/job/corporate_jobs.dm:45]: Not sure what the consensus is on removing those so I'll refrain from commenting, though expect it to probably change.

File: maps/torch/job/outfits/command_outfits.dm [hunk 8]
```diff
@@ -132,4 +132,13 @@
...
+	shoes = /obj/item/clothing/shoes/dutyboots
+
+/decl/hierarchy/outfit/job/torch/crew/command/solrep
+	name = OUTFIT_JOB_NAME("Sol Gov Representative")
+	uniform = /obj/item/clothing/under/rank/internalaffairs/plain/solgov
+	shoes = /obj/item/clothing/shoes/dress
+	suit = /obj/item/clothing/suit/storage/toggle/suit/black
+	id_type = /obj/item/weapon/card/id/torch/crew/adjudicator
+	pda_type = /obj/item/modular_computer/pda/heads
+	l_ear = /obj/item/device/radio/headset/heads/torchexec
```
Comment (Kaksisilma) [inline maps/torch/job/outfits/command_outfits.dm]:
```suggestion
	l_ear = /obj/item/device/radio/headset/bridgeofficer
```

File: maps/torch/torch_areas.dm [hunk 9]
```diff
@@ -682,6 +682,16 @@
...
+	icon_state = "heads_sol"
+	name = "\improper Command - Rep's Office"
+	req_access = list(access_adjudicator)
+
+/area/crew_quarters/heads/office/solrep/backroom
+	icon_state = "heads_sol"
+	name = "\improper Command - Rep's Backroom"
+	req_access = list(access_adjudicator)
+
 /area/crew_quarters/heads/office/vip
```
Comment (Kaksisilma) [inline maps/torch/torch_areas.dm:695]: If this is now an unused area, it can (should) be deleted.

**Review by Kaksisilma — CHANGES_REQUESTED** [review]
Though I don't mind the addition of this new role, currently this feels a little too much like CO 2.0

Also what's up with the APA removal?

**Comment by Kaksisilma** [comment]
Feels like it would be better to just re-convert the VIP Room back to the Sol Rep's room aswell, honestly, though IDK Carl's take on this.

**Comment by Mannybrado** [comment]
I'll fix this stuff tomorrow.

**Comment by ProbablyCarl** [comment]
> Feels like it would be better to just re-convert the VIP Room back to the Sol Rep's room aswell, honestly, though IDK Carl's take on this.

Of the same mind. Retain CL for the moment, move SGRep back into their old office.

**Comment by ProbablyCarl** [comment]
CL Rep removal can happen in a future PR. Would rather it be separate from the one reintroducing the SGRep.

**Comment by Vizoplez** [comment]
Dunno why we're removing APA instead of shifting it to the SGR.

File: code/game/objects/items/weapons/cards_ids.dm (hunk 1 above)
Comment (Mannybrado) [reply code/game/objects/items/weapons/cards_ids.dm]: done

File: maps/torch/job/command_jobs.dm (hunk 2 above)
Comment (Mannybrado) [reply maps/torch/job/command_jobs.dm]: done

File: maps/torch/job/command_jobs.dm (hunk 3 above)
Comment (Mannybrado) [reply maps/torch/job/command_jobs.dm]: done

File: maps/torch/job/command_jobs.dm (hunk 4 above)
Comment (Mannybrado) [reply maps/torch/job/command_jobs.dm]: done

File: maps/torch/job/command_jobs.dm (hunk 5 above)
Comment (Mannybrado) [reply maps/torch/job/command_jobs.dm]: Cut it down significantly. They still have a little bit more but, for instance, its still less than a BO.

File: maps/torch/job/command_jobs.dm (hunk 6 above)
Comment (Mannybrado) [reply maps/torch/job/command_jobs.dm]: yeah... that's better. done.

File: maps/torch/job/outfits/command_outfits.dm (hunk 8 above)
Comment (Mannybrado) [reply maps/torch/job/outfits/command_outfits.dm]: done.

File: maps/torch/torch_areas.dm (hunk 9 above)
Comment (Mannybrado) [reply maps/torch/torch_areas.dm:695]: done.

File: maps/torch/job/corporate_jobs.dm (hunk 7 above)
Comment (Mannybrado) [reply maps/torch/job/corporate_jobs.dm:45]: converted it into the guard for the sol rep.

**Comment by Mannybrado** [comment]
updated.

File: maps/torch/job/command_jobs.dm [hunk 10]
```diff
@@ -369,3 +369,107 @@
...
+							 /datum/computer_file/program/camera_monitor,
+							 /datum/computer_file/program/reports)
+
+	access = list(access_representative, access_security, access_medical,
+					access_emergency_storage, access_change_ids,
+					access_eva, access_bridge,
+					access_janitor, access_cargo,
+					access_hydroponics, access_library, access_network, access_mailsorting,
+					access_hangar, access_aquila, access_solgov_crew,
+					access_commissary,)
```
Comment (Kaksisilma) [inline maps/torch/job/command_jobs.dm]: While I get the need to give SolRep accesses, most of those are completely out of the scope of what they should be doing. Again, they are a bureaucrat and a RP role. 

In order from most to least egregious, **no to:** access_change_ids, access_network access_aquila, access_eva, access_mailsorting, access_comissary

I'm also pretty sure access_library is a relic from bay boxstation era since it hardly appears anywhere else on code, so you can remove it too because it is literally unused.

File: maps/torch/job/command_jobs.dm [hunk 11]
```diff
@@ -369,3 +369,107 @@
...
+	                    SKILL_BUREAUCRACY = SKILL_MAX)
+	alt_titles = list(
+		"Sol Federal Assistant",
+		"Sol Federal Chaperone"	)
+	skill_points = 20
+	access = list(access_representative, access_maint_tunnels, access_security, access_medical,
+						access_engine, access_research, access_bridge,
+						access_cargo, access_solgov_crew, access_hangar,
+						access_commissary, access_aquila,
+						access_sec_guard)
```
Comment (Kaksisilma) [inline maps/torch/job/command_jobs.dm]: Same accesses as the Sol Rep, plus sec_guard for their locker, please.

File: maps/torch/job/torch_access.dm [hunk 12]
```diff
@@ -135,6 +135,13 @@
 	desc = "Commissary"
 	region = ACCESS_REGION_GENERAL
 
+/var/const/access_representative = "ACCESS_REPRESENTATIVE" //97
+/datum/access/representative
+	id = access_representative
+	desc = "Sol Gov Representative"
+	region = ACCESS_REGION_COMMAND
+	access_type = ACCESS_TYPE_NONE //Ruler of their own domain, CO cannot enter
+
```
Comment (Kaksisilma) [inline maps/torch/job/torch_access.dm:144]: This is OK, but if you want to go for code quality, just completely replace adjudicator with rep stuff.

File: maps/torch/structures/closets/misc.dm [hunk 13]
```diff
@@ -96,15 +96,40 @@
...
+obj/structure/closet/secure_closet/solguard
+	name = "\improper sol agent locker"
+	req_access = list(access_representative)
+	closet_appearance = /decl/closet_appearance/secure_closet/sol/two
+
+/obj/structure/closet/secure_closet/solguard/WillContain()
+	return list(
+		/obj/item/device/flash,
+		/obj/item/clothing/accessory/storage/holster/armpit,
+		/obj/item/weapon/gun/projectile/pistol/throwback,
```
Comment (Kaksisilma) [inline maps/torch/structures/closets/misc.dm]: Are you sure this is the right pistol type?

pistol/throwback can't have their ammo printed on a lathe. I'd suggest going with some other revolver/pistol.

The obvious should be stated, so, nothing overly powerful like a hand canon.

File: maps/torch/structures/closets/misc.dm [hunk 14]
```diff
@@ -96,15 +96,40 @@
...
+		/obj/item/weapon/storage/secure/briefcase,
+		/obj/item/clothing/shoes/laceup,
+		/obj/item/clothing/under/rank/internalaffairs/plain,
+		/obj/item/clothing/suit/storage/toggle/suit/black/sfp,
+		/obj/item/clothing/gloves/color/black,
+		/obj/item/clothing/glasses/sunglasses,
+		/obj/item/clothing/glasses/sunglasses/big,
+		/obj/item/device/flashlight/maglight,
+		/obj/item/device/radio/headset/federal,
+		/obj/item/taperoll/police
```
Comment (Kaksisilma) [inline maps/torch/structures/closets/misc.dm]: He's not part of sec. He won't need this.
```suggestion

```

**Review by Kaksisilma — CHANGES_REQUESTED** [review]
There's a lot of things to fix here. I am not fond at all of the idea of the SolRep moving to the CL's office. Preferably, you'd modify the existing Vip office (which used to be the Sol Rep's) in order to accomodate both of them, else this might be confusing and hard to adapt.

At this point, I feel like just straight-up removing APA and not adding the Fed Agent, or simply keeping the APA and not giving the Sol Rep an agent might be the best idea simply for the ease of mapping. but I'll defer to @ProbablyCarl in this specific topic. 

I'm not against the addition of Sol-Rep. In fact, it has always been one of my favorite roles to play. There is simply a TON of sub-optimal stuff in this PR that seems to steer a RP role into a far more active position, which seems to be borderline BO. It's important to remember that you have 0 power over the captain or any crewmember, and that fax machine in your office is your only lifeline to get anything done through Sol-Gov. You are a bureaucrat, a representative, and a diplomat on behalf of Sol-gov, and nothing else. This being said, I'm not sure if I mentioned this in one of the .dm reviews but there is 0 reason for the CL, APA or SolFed to have access to any other radio channel besides command (Except maybe research for CL, but even then I'd call it a bit of a stretch.) Giving them Network and Id-modification accesses is also... Why?

TL;DR:
Fix the accesses
Do not move into the CL office. Map the Vip Office back to Sol-Rep office and give them an extra locker for the SolFed agent.
Why does he need access to Engineering, Sec and Infantry?

My hot take is that neither of them needs a protection agent in a ship with extensive security detail, but It's an enjoyable role to play for some people so I guess it can stay.

File: maps/torch/job/command_jobs.dm (hunk 10 above)
Comment (Mannybrado) [reply maps/torch/job/command_jobs.dm]: understandable, done.

File: maps/torch/job/command_jobs.dm (hunk 11 above)
Comment (Mannybrado) [reply maps/torch/job/command_jobs.dm]: done.

File: maps/torch/job/torch_access.dm (hunk 12 above)
Comment (Mannybrado) [reply maps/torch/job/torch_access.dm:144]: I did this originally, but I didn't want to delete any of the adjudicator stuff just in case. I really don't want to do that in this PR, but I could in the future.

File: maps/torch/structures/closets/misc.dm (hunk 13 above)
Comment (Mannybrado) [reply maps/torch/structures/closets/misc.dm]: I'll give them an extra mag. The federal agent having the 1911 clone is just, like... c'mon. You can always print 10mm and fill the magazines manually.
Since they are lethal rounds, and the gun is cool, them having to put a little more effort into conserving or resupplying their ammo seems like a good trade-off for me.

File: maps/torch/structures/closets/misc.dm (hunk 14 above)
Comment (Mannybrado) [reply maps/torch/structures/closets/misc.dm]: done.

**Comment by Mannybrado** [comment]
> I am not fond at all of the idea of the SolRep moving to the CL's office. Preferably, you'd modify the existing Vip office (which used to be the Sol Rep's) in order to accomodate both of them, else this might be confusing and hard to adapt.

I don't see how. The CL is significantly less important to the ship(and in general), so they get the smaller office. I can see how some people might get confused for a round or two, but this is the case with literally any mapping change that effects something already in game. Its not going to cause some kind of calamity where everyone is running around like headless chickens because a single role has a different office.

Just to be clear, I moved the CL's office into the VIP room, and the SGR office into the CL's now old office. It just... makes sense that the SGR would get the better office and I prefer to focus on things that make sense when mapping.

**Comment by Mannybrado** [comment]
As for the APA... I don't know. Its kind of just, like... I don't know. I would prefer to have them removed, as it makes sense to do so and its also a very shallow role, but I could understand doing some kind of vote or something.

The Fed Agent is meant to be a kind of replacement, but it serves a different general function and is meant to be more engaging for people other than the representative.

File: maps/torch/structures/closets/misc.dm (hunk 13 above)
Comment (Kaksisilma) [reply maps/torch/structures/closets/misc.dm]: Sure, but why would the fed agent have a shoddy pistol instead of a proper solarian pistol? By this logic we should give infantry mosins because they look cool. I don't like the idea of giving the rep a throwback because it's A) fundamentally flawed from an ammo perspective even if you can load it manually, B) Why would you give a federal agent a 600 year old pistol when there are better alternatives, C) It's not only old, it's also shoddy and has four different sprites that are not necessarily a 1911. (1 in 4 chance of getting the correct one.)

**Comment by Kaksisilma** [comment]
> As for the APA... I don't know. Its kind of just, like... I don't know. I would prefer to have them removed, as it makes sense to do so and its also a very shallow role, but I could understand doing some kind of vote or something.
> 
> The Fed Agent is meant to be a kind of replacement, but it serves a different general function and is meant to be more engaging for people other than the representative.

Is Fed Agent any more engaging by being assigned to the Sol Rep instead of the CL?

But I digress. Either way, I'm going to wait for @ProbablyCarl opinion on this one and we'll roll with it. This is OK from a code standpoint, though I would prefer if the throwback wasn't there and if they had their original offices.

File: maps/torch/structures/closets/misc.dm (hunk 13 above)
Comment (Mannybrado) [reply maps/torch/structures/closets/misc.dm]: sure, done.

**Comment by Mannybrado** [comment]
> Is Fed Agent any more engaging by being assigned to the Sol Rep instead of the CL?

Fed Agent has more freedom to do what they like. They can do their own investigations for whatever aside from the Sol Rep, if they want. The Fed isn't a security member, but they are still a federal agent and that affords them some kind of niche gimmick. APA does nothing. They sit in the CL's office and follow them around when they leave... until someone tries to attack the CL which generally only happens every couple of rounds... then they go back to shadowing the CL. If people play Fed like that, sucks, but they at least have the ability to do more.
You may say, "But APA can totally do interesting things!" Okay... like what? I firmly believe the only interesting gimmick as APA is rolling assistant and just... being an actually helpful assistant, yet I am literally the only person that has ever done this. I suppose being a REAL corporate goon and brutalizing people is interesting, but most CLs don't play the kind of person to allow that. Sitting around like a statue waiting for someone to attack the CL is boring and I don't understand how people can do it every round they get the role.

**Comment by ProbablyCarl** [comment]
As per Kak's review:

This PR looks good. I've a few requests if you're willing to make concessions, however.
Retain the bodyguard for CL.
Access might need touching up.


That's about it.

**Comment by Vizoplez** [comment]
> > Is Fed Agent any more engaging by being assigned to the Sol Rep instead of the CL?
> 
> Fed Agent has more freedom to do what they like. They can do their own investigations for whatever aside from the Sol Rep, if they want. The Fed isn't a security member, but they are still a federal agent and that affords them some kind of niche gimmick. APA does nothing. They sit in the CL's office and follow them around when they leave... until someone tries to attack the CL which generally only happens every couple of rounds... then they go back to shadowing the CL. If people play Fed like that, sucks, but they at least have the ability to do more. You may say, "But APA can totally do interesting things!" Okay... like what? I firmly believe the only interesting gimmick as APA is rolling assistant and just... being an actually helpful assistant, yet I am literally the only person that has ever done this. I suppose being a REAL corporate goon and brutalizing people is interesting, but most CLs don't play the kind of person to allow that. Sitting around like a statue waiting for someone to attack the CL is boring and I don't understand how people can do it every round they get the role.

APA is more than just a physical guard for the Liason. They're also supposed to keep NT's; or their employers, interests at heart - beyond the liaison.

This includes cooperating with local security forces, for example, if their target goes rogue; and attempting to non-lethally subdue them/etc (unless you play a corrupt LPA); but as a better note.

The APA/LPA exists to advise and protect the CL. This extends far more than merely physically. I play LPA's that go as far as to give actual legal advice towards my target, because; simply.

You're there to
A) Ensure your employers needs are met within reason
B) Stay alive
C) Keep your target alive, and out of jail, within reason

**Comment by ProbablyCarl** [comment]
No idea as to what happened to Manny.
Going to close this for now.
